src/document.c

```diff
@@ -472,12 +472,15 @@ FIELD_BULK_INDEXER(numericIndexer) {
 }
 
 FIELD_PREPROCESSOR(geoPreprocessor) {
-  // TODO: streamline
-  const char *c = RedisModule_StringPtrLen(field->text, NULL);
-  if (*c == '"') {
-    c += 1;
-  }
-  char *pos = strpbrk(c, " ,");
+  // buffer of 32 bytes should be sufficient. lon,lat 
+  size_t len;
+  char buf[32] = { 0 };
+  const char *str = RedisModule_StringPtrLen(field->text, &len);
+  if (*str == '"') {
+    str += 1;
+  }
+  memcpy(buf, str, MIN(32, len));
```

Comment:
I think we need to ensure '\0` at the end if `buf`.

Reply:
In line 477 we clear the buffer which is equivalent.

Reply:
It does not clear the buffer, just setting the first byte to 0,

Reply:
@ashtul small comment: at line 488 `MIN(32, len)` can be replaced with `len` since we did not return from line 484 so `len <= 31`